File: refrigitz 12/TakeImageMemmorry.cpp
#include "TakeImageMemmorry.h"


namespace GalleryStudio
{

const std::wstring TakeImageMemmorry::S = L"TakeImages.txt";
GalleryStudio::TakeImageMemmorry *TakeImageMemmorry::Node = new TakeImageMemmorry();

	void TakeImageMemmorry::Load()
	{
		try
		{
			if (Node == nullptr)
			{
				Node = new TakeImageMemmorry();
			}
//C# TO C++ CONVERTER WARNING: C# to C++ Converter converted the original 'null' assignment to a call to 'delete', but you should review memory allocation of all pointer variables in the converted code:
			delete Node->getTakeImagesNextAccess();
//C# TO C++ CONVERTER WARNING: C# to C++ Converter converted the original 'null' assignment to a call to 'delete', but you should review memory allocation of all pointer variables in the converted code:
			delete Node->getTakeImagesCurrentAccess();

			FileStream *DummyFileStream = new FileStream(S, System::IO::FileMode::OpenOrCreate, System::IO::FileAccess::ReadWrite);
			BinaryFormatter *Formatters = new BinaryFormatter();
			GalleryStudio::TakeImage *Dummy = new TakeImage(0, nullptr);
			GalleryStudio::TakeImageMemmorry *Last = nullptr;
			std::cout << std::wstring(L"Loading...") << std::endl;
			DummyFileStream->Seek(0, SeekOrigin::Begin);
			while (DummyFileStream->Position < DummyFileStream->Length)
			{

				Dummy = static_cast<TakeImage*>(Formatters->Deserialize(DummyFileStream));
				if (Node->Current == nullptr)
				{
					Node->Current = Dummy;
				}
				else
				{
					Last = Node;
					while (Last->Next != nullptr)
					{
						Last = Last->Next;
					}
					TakeImageMemmorry *New = new TakeImageMemmorry();
					New->Current = Dummy;
					Last->setTakeImagesNextAccess(New);
				}
			}
			DummyFileStream->Flush();
			DummyFileStream->Close();
		}
		catch (IOException t)
		{
//C# TO C++ CONVERTER TODO TASK: There is no native C++ equivalent to 'ToString':
			std::cout << t->Message->ToString() << std::endl;
		}
	}

	void TakeImageMemmorry::DeleteObject(TakeImageMemmorry *p)
	{
		TakeImageMemmorry *t = new TakeImageMemmorry();
		t = Node;
		try
		{
			if (static_cast < t::getTakeImagesCurrentAccess()::getCustomersAcess()*>(!) = (p->getTakeImagesCurrentAccess()->getCustomersAcess()))
			{
				if (t != nullptr)
				{
					while (static_cast < t::getTakeImagesNextAccess()::getTakeImagesCurrentAccess()::getCustomersAcess()*>(!) = (p->getTakeImagesCurrentAccess()->getCustomersAcess()))
					{
						if (t->getTakeImagesNextAccess() != nullptr)
						{
							t = t->getTakeImagesNextAccess();
						}
						else
						{
							if (static_cast < t::getTakeImagesCurrentAccess()::getCustomersAcess()*>(!) = (p->getTakeImagesCurrentAccess()->getCustomersAcess()))
							{
//C# TO C++ CONVERTER WARNING: C# to C++ Converter converted the original 'null' assignment to a call to 'delete', but you should review memory allocation of all pointer variables in the converted code:
								delete t;
								break;
							}
						}
					}
				}
				if (t != nullptr)
				{
					if (t->getTakeImagesNextAccess() != nullptr)
					{
						t->setTakeImagesNextAccess(t->getTakeImagesNextAccess()->getTakeImagesNextAccess());
					}

					else
					{
//C# TO C++ CONVERTER WARNING: C# to C++ Converter converted the original 'null' assignment to a call to 'delete', but you should review memory allocation of all pointer variables in the converted code:
						delete t->getTakeImagesNextAccess();
					}
				}

			}
			else
			{
				t = t->getTakeImagesNextAccess();
				Node = t;
			}
		}
		catch (NullReferenceException q)
		{
//C# TO C++ CONVERTER TODO TASK: There is no native C++ equivalent to 'ToString':
			std::cout << q->Message->ToString() << std::endl;

		}
	}

	void TakeImageMemmorry::AddObject(TakeImageMemmorry *p)
	{
		//p.TakeImagesNextAccess = null;
		TakeImageMemmorry *t = new TakeImageMemmorry();
		t = p->getTakeImagesNodeAccess();
		while (t->getTakeImagesNextAccess() != nullptr)
		{
			t = t->getTakeImagesNextAccess();
		}
		 if (t->getTakeImagesCurrentAccess() == nullptr)
		 {
			t->setTakeImagesCurrentAccess(p->getTakeImagesCurrentAccess());
		 }
		 else
		 {
		 t->setTakeImagesNextAccess(p);
		 }

	}

	TakeImageMemmorry *TakeImageMemmorry::getTakeImagesNodeAccess() const
	{
		return Node;
	}

	void TakeImageMemmorry::setTakeImagesNodeAccess(TakeImageMemmorry *value)
	{
		Node = value;
	}

	TakeImage *TakeImageMemmorry::getTakeImagesCurrentAccess() const
	{
		return Current;
	}

	void TakeImageMemmorry::setTakeImagesCurrentAccess(TakeImage *value)
	{
		Current = value;
	}

	TakeImageMemmorry *TakeImageMemmorry::getTakeImagesNextAccess() const
	{
		return Next;
	}

	void TakeImageMemmorry::setTakeImagesNextAccess(TakeImageMemmorry *value)
	{
		Next = value;
	}

	void TakeImageMemmorry::InitializeInstanceFields()
	{
		Current = new TakeImage(0,nullptr);
		Next = 0;
	}
}

File: GalleryStudio/TakeImageMemmorry.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Runtime.Serialization.Formatters.Binary;
using System.Runtime.Serialization;
using System.IO;
namespace GalleryStudio
{

    public class TakeImageMemmorry
    {
        const string S = "TakeImages.txt";
        static GalleryStudio.TakeImageMemmorry Node = new TakeImageMemmorry();
        GalleryStudio.TakeImage Current = new TakeImage(0,null);
        GalleryStudio.TakeImageMemmorry Next = null;

        public void Load()
        {
            try
            {
                if (Node == null)
                    Node = new TakeImageMemmorry();
                Node.TakeImagesNextAccess = null;
                Node.TakeImagesCurrentAccess = null;

                FileStream DummyFileStream = new FileStream(S, System.IO.FileMode.OpenOrCreate, System.IO.FileAccess.ReadWrite);
                BinaryFormatter Formatters = new BinaryFormatter();
                GalleryStudio.TakeImage Dummy = new TakeImage(0, null);
                GalleryStudio.TakeImageMemmorry Last = null;
                Console.WriteLine("Loading...");
                DummyFileStream.Seek(0, SeekOrigin.Begin);
                while (DummyFileStream.Position < DummyFileStream.Length)
                {

                    Dummy = (TakeImage)Formatters.Deserialize(DummyFileStream);
                    if (Node.Current == null)
                    {
                        Node.Current = Dummy;
                    }
                    else
                    {
                        Last = Node;
                        while (Last.Next != null)
                            Last = Last.Next;
                        TakeImageMemmorry New = new TakeImageMemmorry();
                        New.Current = Dummy;
                        Last.TakeImagesNextAccess = New;
                    }
                }
                DummyFileStream.Flush();
                DummyFileStream.Close();
            }
            catch (IOException t)
            {
                Console.WriteLine(t.Message.ToString());
            }
        }
        public void DeleteObject(TakeImageMemmorry p)
        {
            TakeImageMemmorry t = new TakeImageMemmorry();
            t = Node;
            try
            {
                if ((t.TakeImagesCurrentAccess.CustomersAcess) != (p.TakeImagesCurrentAccess.CustomersAcess))
                {
                    if (t != null)
                        while ((t.TakeImagesNextAccess.TakeImagesCurrentAccess.CustomersAcess) != (p.TakeImagesCurrentAccess.CustomersAcess))
                        {
                            if (t.TakeImagesNextAccess != null)
                                t = t.TakeImagesNextAccess;
                            else
                                if ((t.TakeImagesCurrentAccess.CustomersAcess) != (p.TakeImagesCurrentAccess.CustomersAcess))
                                {
                                    t = null;
                                    break;
                                }
                        }
                    if (t != null)
                    {
                        if (t.TakeImagesNextAccess != null)
                            t.TakeImagesNextAccess = t.TakeImagesNextAccess.TakeImagesNextAccess;

                        else
                            t.TakeImagesNextAccess = null;
                    }

                }
                else
                {
                    t = t.TakeImagesNextAccess;
                    Node = t;
                }
            }
            catch (NullReferenceException q)
            {
                Console.WriteLine(q.Message.ToString());
            
            }
        }
        public void AddObject(TakeImageMemmorry p)
        {
            //p.TakeImagesNextAccess = null;
            TakeImageMemmorry t =new TakeImageMemmorry();
            t = p.TakeImagesNodeAccess;
            while (t.TakeImagesNextAccess != null)
                t = t.TakeImagesNextAccess;
             if (t.TakeImagesCurrentAccess == null)
                t.TakeImagesCurrentAccess = p.TakeImagesCurrentAccess;
             else
             t.TakeImagesNextAccess = p;
        
        }
        public TakeImageMemmorry TakeImagesNodeAccess
        {
            get
            {
                return Node;
            }
            set
            {
                Node = value;
            }
        }
        public TakeImage TakeImagesCurrentAccess
        {
            get
            {
                return Current;
            }
            set
            {
                Current = value;
            }
        }
        public TakeImageMemmorry TakeImagesNextAccess
        {
            get
            {
                return Next;
            }
            set
            {
                Next = value;
            }
        }
    }
    }
